### ISSUE TITLE
Auto Wrap csharp three modes 

### ISSUE THREAD

**Issue by Laeeth** [symmetryinvestments/autowrap#117]
Typically you either want to wrap custom code and if it fails then fail hard.  Or you want to do best efforts on a large piece of code you don't control and then progressively work away at the rest with failures reported as warnings at compile time and the generated c sharp being able to compile if it's missing some things. 

A further possibility is you want to use static foreach just to wrap some types and functions in a module.  So on manual control. 

At the moment auto wrap is only hard fail mode for csharp.  Could we allow the other modes too please.  See SIL for example usage.  In stdlib directory.

**Comment by atilaneves** [comment]
> A further possibility is you want to use static foreach just to wrap some types and functions in a module. So on manual control.

Normally done by using `export` on the wanted ones.

**Comment by Laeeth** [comment]
I don't see how you can depend on sticking export on the wanted ones if you don't control or don't fully control the code you are wrapping.  So it's like saying to a guy who walks to work and wants to drive its normally done by walking.  I know but I'd like to get a car.

If you look at the SIL stdlib that's a good illustration of a representative use case.

**Comment by atilaneves** [comment]
> if you don't control or don't fully control the code you are wrapping

`export` won't work in that situation, no. I wanted to point out that it's a way of achieving control over what gets wrapped if it's an available option. I understand the need for finer grained control.

**Comment by Laeeth** [comment]
Yes, I suppose quite often you state what seems like it should be obvious in the context as if it's something I might not know,and I don't understand why.  If I recall correctly I wrote the first version of autowrap for SIL (heavily pirating your code).  I don't know how someone could do that without understanding that yes of course export is a possibility.  Usually if I ask for something it's not without some awareness of the context and maybe I could be wrong but it's not unthoughtful.
